File: rmvp-annotations/src/main/java/com/xfinity/rmvp_annotations/DefaultPresenter.java
package com.xfinity.rmvp_annotations;

import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.TYPE;

@Target(value = TYPE)
public @interface DefaultPresenter {
    Class<?> viewClass();

    //TODO if the 'library' project were a straight java project, we could enforce stricter rules on this annotation,
    // i.e. Class<? extends ComponentView> viewClass();
}

File: blueprint-annotations/src/main/java/com/xfinity/blueprint_annotations/DefaultPresenter.kt
package com.xfinity.blueprint_annotations

import kotlin.reflect.KClass

@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
annotation class DefaultPresenter(val viewClass: KClass<*>)

//TODO if the 'library' project were a straight java project, we could enforce stricter rules on this annotation,
// i.e. val viewClass: KClass<out ComponentView>
